### ISSUE TITLE
Menu link - right click

### ISSUE THREAD

**Issue by CIMAManuel** [GFDRR/open-risk-data-dashboard#168]
Il menu e i link all'interno della piattaforma sono gestiti tramite il modulo $locationProvider.html5Mode di Angular JS. Questo modulo impedisce di fatto di avere dei link "html standard" e di conseguenza un banale right click sulla voce di menu non funziona ottenendo un link del tipo "frontend/%7B%7B$locationProvider.$get%7D%7D". Se questi link sono trasformati in tag a standard il click viene riconosciuto (la barra degli indirizzi si aggiorna) ma il contenuto non viene caricato.

Esempio di link del menu attuale: `<li><a href="contribute.html" ng-click="changeview('contribute.html', '1')">Submit a dataset</a></li>
`
Per risolvere questo bug le possibili soluzioni identificate sono le seguenti:
1. Eliminare il modulo $locationProvider: non utilizzare questo modulo di Angular riporterebbe la gestione dei tag A al normale funzionamento.
Pro: link standard anche nel menu di navigazione
Contro: occorre fare un check completo di tutti i file HTML e JS del progetto per modificare il parametro della variabile $location (path / search) e disabilitare il $locationprovider

2. Abbiamo identificato che se nel link del menu oltre alla funzione Angular "changeview" si imposta anche il valore HREF del tag A il link interpreta sia la chiamata angular che quella standar HTML ottenendo così la risoluzione del "right click".
Pro: si risolve il problema relativo al right click
Contro: occorre identificare dove è utilizzata la funzione changeview e intervenire sui link aggiungendo il relativo valore nel tag href. Capire la compatibilità con tutti i browser
**************************************************************************************************************
The platform menu/links are manage by Angular modul $locationProvider.html5Mode.
This modul force to use the menu links only with angular function and the links are not "html standard".
In this conditions if you do a "right click - open in new tab" over one menu link you get a 404 error because the link you get is like this: "...frontend/%7B%7B$locationProvider.$get%7D%7D".
Otherwise if you set a standard html tag A in the menu options the browser update the link url but it doesn't load the new content.

Link example: "`<a href="" ng-click="changeview('methodology.html', '2')">Documentation</a>`".

We have identified two solutions:

1. Delete Angular modul $locationProvider: not using this Angular module would report the management of the A tags to normal operation.
Pro: standard link also in the navigation menu
Cons: you must check all the HTML and JS files of the project to modify the parameter of the variable $location (path / search) and disable the $ locationprovider

2. We have identified that if in the menu link, in addition to the Angular "changeview" function, the HREF value of the tag A is set, the link interprets both the angular call and the standar HTML call, thus obtaining 
the "right click" resolution.
Pro: HTML thus obtaining the resolution of the "right click".
Cons: it is necessary to identify where the changeview function is used and to intervene on the links by adding the relative value in the href tag. Understanding compatibility with all browsers

**Comment by pzwsk** [comment]
Thanks, I saw that the use of Angular is causing several issues,

**Comment by oncletom** [comment]
Yes. It might be worth opening a separate issue to describe this filtering problem. I suspect it is bound to a different aspect of Angular.

About this issue, we decided to go with solution 1 because it requires the lowest amount of effort and present a lower risk. All clickable links will have a real `href` attribute which is clickable, shareable and indexable by search engines.

**Comment by CIMAManuel** [comment]
With solution 2 I found a problem with navigation.
The right click and the link are ok but if you want to come back in previus page the "back button" of browser doesn't work fine, the url change but the page is not refresh.
What do you think about that?
@pzwsk and @oncletom

**Comment by oncletom** [comment]
@CIMAManuel I don't know, can you open a pull request with the code changes so as I can test and look for solutions?

**Comment by oncletom** [comment]
I don't know if it is the same problem as the "right click" issue per say, but I encountered navigation issue during a user interview.

When you click on "Explore Countries", then "Slovenia", then "Aerial imagery". Then try to hit the "back button" of the web browser. There are 2 scenarios:

1. if you click on links, you will be brought back to the homepage 
<img width="213" alt="image" src="https://user-images.githubusercontent.com/138627/52044091-8a020180-2541-11e9-82de-3a6ae30158cc.png">
2. if you click on the row, you will be brought back to the previous page (expected behaviour)
<img width="294" alt="image" src="https://user-images.githubusercontent.com/138627/52044121-a140ef00-2541-11e9-9160-8414c3eb82fb.png">

**Comment by oncletom** [comment]
Maybe one solution is to disable `rewriteLinks`, thus, configuring `app.js` with this instead: 

```js
$locationProvider.html5Mode({
  enabled: true,
  rewriteLinks: false
});
```

**Comment by oncletom** [comment]
Fixed by #425
